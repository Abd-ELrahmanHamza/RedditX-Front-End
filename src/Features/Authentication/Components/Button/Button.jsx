import { BaseButton } from "./Button.styled";

const Button = ({ children, ...otherProps }) => {
  return <BaseButton {...otherProps}>{children}</BaseButton>;
};

export default Button;
