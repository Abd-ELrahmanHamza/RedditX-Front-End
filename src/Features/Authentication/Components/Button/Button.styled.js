import styled from "styled-components";

const logInButtonColor = "#d93a00";
const logInButtonColorHovered = "#dc4a14";
const logInBlue = "#0079d3";
const logInBlueHovered = "#3394dc";
const fontFamily = "'IBM Plex Sans', sans-serif;";

export const BaseButton = styled.button`
  line-height: 40px;
  background-color: ${(p) => (p.page ? logInBlue : logInButtonColor)};
  border: none;
  cursor: pointer;
  display: flex;
  justify-content: center;
  font-family: Noto Sans, sans-serif;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: unset;
  border-radius: ${(p) => (p.page ? "5px" : "999px")};
  color: #fff;
  height: 40px;
  padding: 0 16px;
  font-family: ${fontFamily};
  text-transform: unset;
  margin-top: 8px;
  width: 100%;

  opacity: ${(p) => (p.valid || p.page ? "1" : "0.3")};

  cursor: ${(p) => (p.valid || p.page ? "pointer" : "default")};

  &:hover {
    background-color: ${(p) =>
      !p.page
        ? p.valid
          ? logInButtonColorHovered
          : logInButtonColor
        : logInBlueHovered};
  }
`;
