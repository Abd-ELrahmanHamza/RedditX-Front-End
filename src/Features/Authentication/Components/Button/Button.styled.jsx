import styled from "styled-components";

const logInButtonColor = "#d93a00";
const logInButtonColorHovered = "#dc4a14";
const fontFamily = "'IBM Plex Sans', sans-serif;";

export const BaseButton = styled.button`
  line-height: 40px;
  background-color: ${logInButtonColor};
  border: none;
  cursor: pointer;
  display: flex;
  justify-content: center;
  font-family: Noto Sans, sans-serif;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: unset;
  border-radius: 999px;
  color: #fff;
  height: 40px;
  padding: 0 16px;
  font-family: ${fontFamily};
  text-transform: unset;
  margin-top: 8px;
  width: 100%;

  opacity: ${p => (p.valid ? "1" : "0.3")};

  cursor: ${p => (p.valid ? "pointer" : "default")};

  &:hover {
    background-color: ${p => (p.valid ? logInButtonColorHovered : logInButtonColor)};
  }
`;
