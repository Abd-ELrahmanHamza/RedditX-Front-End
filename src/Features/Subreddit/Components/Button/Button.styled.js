import styled from "styled-components";

export const StyledButton = styled.div`
  display: flex;


  .width {
    width: 96px;
  }

  button {
    width: 100%;
    position: relative;
    background-color: #0079d3;
    border: none;
    color: #fff;
    fill: #fff;
    font-family: "Noto Sans", Arial, sans-serif;
    font-size: 14px;
    font-weight: 700;
    letter-spacing: unset;
    line-height: 17px;
    text-transform: unset;
    min-height: 32px;
    min-width: 32px;
    padding: 4px 16px;
    border-radius:100px;
  }

  button:hover{
    background-color: #1483d6;
  }

  /* button::before {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    border-radius: 9999px;
    background: #fff;
    opacity: 0;
    font-family: "Noto Sans", Arial, sans-serif;
    font-size: 14px;
    font-weight: 700;
    letter-spacing: unset;
    line-height: 17px;
    text-transform: unset;
  } */
`;
