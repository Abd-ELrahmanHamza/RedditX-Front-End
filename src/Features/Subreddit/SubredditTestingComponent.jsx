import { ThemeProvider } from "styled-components";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";
import { BrowserRouter as Router } from "react-router-dom";
import { SubRedditProvider } from "Features/Subreddit/Contexts/SubRedditProvider";
import { SubRedditIDProvider } from "./Contexts/SubRedditIDProvider";
import { IsModeratorProvider } from "Features/Subreddit/Contexts/IsModeratorProvider";
import { IsBannedProvider } from "Features/Subreddit/Contexts/IsBannedProvider";
import { IsMutedProvider } from "Features/Subreddit/Contexts/IsMutedProvider";

const theme = { ...defaultTheme, ...lightTheme };

/**
 * Function that wraps Children props inside a Browser Router and a Theme Provider
 * @returns {React.Component}The component is being returned.
 */
const SubredditTestingComponent = ({ children }) => {
  return (
    <Router>
      <ThemeProvider theme={theme}>
        <SubRedditProvider>
          <SubRedditIDProvider>
            <IsModeratorProvider>
              <IsBannedProvider>
                <IsMutedProvider>{children}</IsMutedProvider>
              </IsBannedProvider>
            </IsModeratorProvider>
          </SubRedditIDProvider>
        </SubRedditProvider>
      </ThemeProvider>
    </Router>
  );
};

export default SubredditTestingComponent;
