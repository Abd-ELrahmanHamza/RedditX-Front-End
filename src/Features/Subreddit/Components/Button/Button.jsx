import { useState } from "react";
import { StyledButton } from "./Button.styled";

const Button = (props) => {

  const [text, setText] = useState(props.isJoined?"join":"joined");
  const onJoin = () =>{
    setText((prev)=>(prev==="join")?"joined":"join")
  }
  const onHover = () => {
    if(props.isJoined){
      setText("leave")
    }
  }

  const onLeave = () =>{
    if(props.isJoined){
      setText("joined")
    }
  }

  return ( 
    <StyledButton>
      <div className="width">
        <button onClick={onJoin} onMouseOver={onHover} onMouseOut={onLeave}>{text}</button>
      </div>
    </StyledButton>
   );
}
 
export default Button;