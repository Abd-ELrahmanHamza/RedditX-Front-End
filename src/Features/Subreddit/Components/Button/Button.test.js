import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Button from "./Button";
import SubredditTestingComponent from "Features/Subreddit/SubredditTestingComponent";

function onJoin() {
  console.log("click Join");
}

describe("Join and Leave button", () => {
  it("should have text join", () => {
    //arrange
    render(
      <SubredditTestingComponent>
        <Button isJoined={false} onJoin={onJoin} />
      </SubredditTestingComponent>
    );

    //assert
    const buttonElement = screen.getByText(/join/i);
    expect(buttonElement).toBeInTheDocument();
  });

  it("text should change", () => {
    //arrange
    render(
      <SubredditTestingComponent>
        <Button isJoined={true} onJoin={onJoin} />
      </SubredditTestingComponent>
    );

    //act
    const button = screen.getByRole("button");
    userEvent.click(button);

    //assert
    const buttonElement = screen.getByText(/joined/i);
    expect(buttonElement).toBeInTheDocument();
    // expect(button).toHaveTextContent("join");

    userEvent.hover(button);
    const leaveElement = screen.getByText(/leave/i);
    expect(leaveElement).toBeInTheDocument();
  });
});
