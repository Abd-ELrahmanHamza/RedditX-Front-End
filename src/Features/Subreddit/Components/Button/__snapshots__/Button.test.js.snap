// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join and Leave button should render without crashing 1`] = `
<styled.div>
  <div
    className="width"
  >
    <button
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      join
    </button>
  </div>
</styled.div>
`;
